### ISSUE TITLE
how to prevent the detected object overlap together?

### ISSUE THREAD

**Issue by hebo1982** [ultralytics/yolov5#9029]
I train model with "mnist" dataset.
But some digit recognize as two digit (like 7 recognize 7 and 1).
How to prevent like this?

![BFA39B06F5F6F1CB129EC20E895587E8](https://user-images.githubusercontent.com/5362672/185541018-5ded1543-790d-494b-840f-53801caecd1d.jpg)
<img width="1312" alt="image" src="https://user-images.githubusercontent.com/5362672/185541280-30c0f4ad-554e-4357-b434-a83208b5edf8.png">

**Comment by MartinPedersenpp** [comment]
Have you tried reducing the IOU on the inference?

**Comment by hebo1982** [comment]
@MartinPedersenpp 

I tried reduce iou thresold, but overlap still exist.
I have no idea~

`iou_thres=0.01  # NMS IOU threshold`

**Comment by MartinPedersenpp** [comment]
@hebo1982 
Then I think all you can do is continue training either until you start seeing overfitting or try evolving your hyperparameters before starting a new training session.
Alternatively: Depending on your use case, you can check the confidence score on the "7"s detected as "1"s and vice versa, if the one with the highest confidence is correct, then you can just create a script to fetch the one with the highest confidence.
I hope this helps

**Comment by hebo1982** [comment]
@MartinPedersenpp 

Thanks for your reply. I'll try these.

**Comment by hebo1982** [comment]
@MartinPedersenpp 

`pred = non_max_suppression(pred, conf_thres, iou_thres, classes, agnostic_nms, max_det=max_det)`

In this code  set agnostic_nms = true and increase "iou_thres" that work well!
But I need to understand this part of the code.

**Comment by MartinPedersenpp** [comment]
> @MartinPedersenpp
> 
> `pred = non_max_suppression(pred, conf_thres, iou_thres, classes, agnostic_nms, max_det=max_det)`
> 
> In this code set agnostic_nms = true and increase "iou_thres" that work well! But I need to understand this part of the code.

Hello @hebo1982 agnostic_nms means that when the network is doing non max suppresion, it looks at all labels as being the same, so I will keep the one with the highest confidence when doing nms and iou.
Depending on the confidence of the prediction, you could also increase the confidence threshold to remove the false positives.

**Comment by hebo1982** [comment]
@MartinPedersenpp 
I get it!
Thanks for your help, I close the issue.
